src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -452,6 +457,299 @@ public void testIsNameVisibleTo() {
+        // ins2 -> ins team
+        FeedbackResponseAttributes fr191 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ9S1C1");
+        // team1 -> team2
+        FeedbackResponseAttributes fr211 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ1S2C1");
+        // stu4 -> stu3
+        FeedbackResponseAttributes fr221 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ2S2C1");
+        // ins1 -> stu3
+        FeedbackResponseAttributes fr231 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ3S2C1");
+        // ins6 -> stu5
+        FeedbackResponseAttributes fr232 = responseVisibilityBundle.feedbackResponses.get("FRV.response2ForQ3S2C1");
+        // stu1 -> ins1
+        FeedbackResponseAttributes fr241 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ4S2C1");
+        // stu1 -> ins4
+        FeedbackResponseAttributes fr242 = responseVisibilityBundle.feedbackResponses.get("FRV.response2ForQ4S2C1");
+        // stu6 -> team3 (own team)
+        FeedbackResponseAttributes fr251 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ5S2C1");
+        // stu5 -> team2 (own team)
+        FeedbackResponseAttributes fr252 = responseVisibilityBundle.feedbackResponses.get("FRV.response2ForQ5S2C1");
+        // stu1 -> team3
+        FeedbackResponseAttributes fr261 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ6S2C1");
+        // team1 -> stu7
+        FeedbackResponseAttributes fr271 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ7S2C1");
+        // team3 -> team1
+        FeedbackResponseAttributes fr281 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ8S2C1");
+        // stu8 -> team1
+        FeedbackResponseAttributes fr291 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ9S2C1");
+        // ins2 -> team2
+        FeedbackResponseAttributes fr201 = responseVisibilityBundle.feedbackResponses.get("FRV.response1ForQ10S2C1");
+
+        Set<String> studentsEmailInTeam1 = new HashSet<>(
+                Arrays.asList(student1.getEmail(), student2.getEmail(), student3.getEmail(), student4.getEmail()));
+        Set<String> studentsEmailInTeam2 = new HashSet<>(Arrays.asList(student5.getEmail()));
+        Set<String> studentsEmailInTeam3 = new HashSet<>(Arrays.asList(student6.getEmail(), student7.getEmail()));
+        Set<String> studentsEmailInTeam4 = new HashSet<>(Arrays.asList(student8.getEmail()));
+
+        ______TS("test if visible to giver");
+
+        assertTrue(frLogic.isResponseVisibleForUser(student1.getEmail(), false, student1, studentsEmailInTeam1,
+                fr111, fq11, null));
+        assertTrue(frLogic.isResponseVisibleForUser(instructor1.getEmail(), true, null, null,
+                fr131, fq13, instructor1));
+        assertTrue(frLogic.isResponseVisibleForUser(instructor1.getEmail(), true, null, Collections.emptySet(),
```

Comment:
Rather than calling `Collections.emptySet` everywhere, just define once in a local variable and use it in all places?

Reply:
Sure